File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int sum=0,n=nums.length,i=0,j=0,m=Integer.MAX_VALUE;
        while(j<n){
            sum+=nums[j];
            while(i<=j && sum>=target){
                m=Math.min(j-i+1,m);
                sum-=nums[i++];
            }
            //m=Math.min(j-i+1,m);
            j++;
        }
        return m==Integer.MAX_VALUE?0:m;
    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        i=j=0
        m=sys.maxint
        sum=0
        while j<len(nums):
            sum+=nums[j]
            while sum>=target and i<=j:
                m=min(m,j-i+1)
                sum-=nums[i]
                i+=1
           
            j+=1
        return 0 if m==sys.maxint else m        